.github/workflows/test.yml

```diff
@@ -184,14 +200,15 @@ jobs:
           key: spec-test-data-${{ hashFiles('packages/validator/test/spec/params.ts') }}
 
       - name: Unit tests
-        # id: unit_tests
-        # Rever to "yarn test:unit" when finished debugging core dumps
-        # run: sudo sh -c "ulimit -c unlimited && /usr/bin/node-with-debug $(which yarn) test:unit"
-        run: yarn test:unit
-      
-      # # Remove when finished debugging core dumps
-      # - uses: './.github/actions/core-dump'
-      #   if: ${{ failure() && steps.unit_tests.conclusion == 'failure' }}
+        id: unit_tests
+        run: |
+          if [ "${{ env.NODE_DEBUG }}" == "true" ]; then
+            sudo sh -c "ulimit -c unlimited"
+          fi
+          yarn test:unit
+
+      - uses: './.github/actions/core-dump'
+        if: ${{ always() && env.NODE_DEBUG == 'true' && steps.unit_tests.conclusion == 'failure' }}
```

Comment:
I'm wondering if there could be a way to integrate the `core-dump` as part of the previous step script. It would help with readability.

Reply:
Not a bad idea but it needs to run as the last step of a job so that it can catch failures from any preceding step.  Ideally this action should be added at the end of any job, as the last step, that may potentially throw a native fault